app/src/main/java/com/android/streetworkapp/model/preferences/PreferencesRepositoryDataStore.kt

```diff
@@ -0,0 +1,161 @@
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import java.io.IOException
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+
+class PreferencesRepositoryDataStore(context: Context) : PreferencesRepository {
+
+  private val Context.dataStore: DataStore<Preferences> by
+      preferencesDataStore(name = "preferences")
+  private val dataStore = context.dataStore
+
+  companion object {
+    val IS_LOGGED_IN = booleanPreferencesKey("is_logged_in")
+    val SAVED_UID = stringPreferencesKey("saved_uid")
+    val SAVED_NAME = stringPreferencesKey("saved_name")
+    val SAVED_SCORE = intPreferencesKey("saved_score")
+
+    private const val IO_ERR = "An IO exception occurred, emitting empty preferences."
+  }
+
+  /** A flow that emits the login state of the user. */
+  val loginStateFlow: Flow<Boolean> =
+      dataStore.data
+          .catch { exception ->
+            // Handle exceptions
+            if (exception is IOException) {
+              emit(emptyPreferences())
+              Log.d("DataStoreManager", IO_ERR)
+            } else {
+              Log.d("DataStoreManager", exception.message.toString())
+            }
+          }
+          .map { preferences -> preferences[IS_LOGGED_IN] ?: false }
+
+  /** A flow that emits the user's uid. */
+  val uidFlow: Flow<String> =
+      dataStore.data
+          .catch { exception ->
+            // Handle exceptions
+            if (exception is IOException) {
+              emit(emptyPreferences())
+              Log.d("DataStoreManager", IO_ERR)
+            } else {
+              Log.d("DataStoreManager", exception.message.toString())
+            }
+          }
+          .map { preferences -> preferences[SAVED_UID] ?: "" }
+
+  /** A flow that emits the user's name. */
+  val nameFlow: Flow<String> =
+      dataStore.data
+          .catch { exception ->
+            // Handle exceptions
+            if (exception is IOException) {
+              emit(emptyPreferences())
+              Log.d("DataStoreManager", IO_ERR)
+            } else {
+              Log.d("DataStoreManager", exception.message.toString())
+            }
+          }
+          .map { preferences -> preferences[SAVED_NAME] ?: "" }
+
+  /** A flow that emits the user's score. */
+  val scoreFlow: Flow<Int> =
+      dataStore.data
+          .catch { exception ->
+            // Handle exceptions
+            if (exception is IOException) {
+              emit(emptyPreferences())
+              Log.d("DataStoreManager", "An IO exception occurred, emitting empty preferences.")
+            } else {
+              Log.d("DataStoreManager", exception.message.toString())
```

Comment:
Suggestion: Do a crtl f and be sure that you use the cst `DataStoreManager` everywhere